pkg/plan_modifier/phase.go

```diff
@@ -26,25 +30,33 @@ func (m customPhaseForUnknownModifier) MarkdownDescription(_ context.Context) st
 
 // PlanModifyString implements the plan modification logic.
 func (m customPhaseForUnknownModifier) PlanModifyString(ctx context.Context, req planmodifier.StringRequest, resp *planmodifier.StringResponse) {
-	// Do nothing if there is no state value.
-	if req.StateValue.IsNull() {
-		return
+	// req.StateValue.IsNull() means it is creating
+	// req.StateValue here is from the read
+	if !req.StateValue.IsNull() {
+		if !strings.Contains(req.StateValue.String(), models.PHASE_HEALTHY) && !strings.Contains(req.StateValue.String(), models.PHASE_PAUSED) {
+			resp.Diagnostics.AddError("Cluster not ready for update operations", "Cluster not in healthy state for update operations please wait...")
```

Comment:
we may also check the condition if phase == models.PHASE_PAUSED, and notify user that he/she needs to(may want to) set pause = false in order to make the cluster healthy

Reply:
I removed this from the phase and put the check here https://github.com/EnterpriseDB/terraform-provider-biganimal/blob/881dd62be9110634381631a0739c55eb0a26aaa6/pkg/provider/resource_cluster.go#L584

I found it is better to do the check on the resource controller layer rather than the plan_modifier layer because if they want to see the plan and not modifier anything but state is not "healthy" or "paused" it will throw an error before the diff appears